### PR TITLE
Update web3 transaction hash from RLP

### LINKED ISSUE #176: Strange hash in transaction obj
__System info:__ MacOS Catalina
[commit 3ac6359](https://github.com/ChainSafe/ethermint/commit/3ac6359e1403730fdf6a9807d04fbb29f567ed99)

__Steps to reproduce:__

1. run eth_getTransaction with params:
**"0xca93358f027c5e79fbc00290ddb96c03c1d4f3fda9a0394c8d637aff5b17b9e5"** (or some present tx in your ethermint chain)

__Expected behavior:__ Hash in transaction obj is the same as hash in parameters

__Actual behavior:__ result of request:
```
{
      blockHash: '0xbb9d1d20f0027f16f8d520611527f6ced36e3d6a019eccccc9e570e389f5c14a',
      blockNumber: 3335,
      from: '0xCC46DC7b409E96c4601228401D8C5Bd74a6951D5',
      gas: 62291,
      gasPrice: '20',
      hash: '0xa74649e1055ba3707910b118bedf999042a82647a8c81052d0ae0c05bddf8d19',
      input: '0xd46e8dd67c5d32be8d46e8dd67c5d32be8058bb8eb970870f072445675058bb8eb970870f072445675',
      nonce: 3,
      to: '0xC4B817c2495FdAF1033e80BF59F2da753400Dbbd',
      transactionIndex: 0,
      value: '145',
      v: '0xe0',
      r: '0x93b9e90f5e8d4e6bd4503539feeb8d80dbc3a42bc8bfa42993db812c084872fc',
      s: '0x3ac6e14db8dcd42a240e23a13e5eb264853421b10ed8fef919c52db601a0c854'
}
```

__Additional info:__ When i go through debug into code, i saw that in [func bytesToEthTx 
 eth_api.go#590](https://github.com/ChainSafe/ethermint/blob/51adade59f2bb9fe498334f74fb83f4886fa7977/rpc/eth_api.go#L590) return `ethTx` with `nil` hash and seems like result hash [eth_api.go#611](https://github.com/ChainSafe/ethermint/blob/51adade59f2bb9fe498334f74fb83f4886fa7977/rpc/eth_api.go#L611) does not calculating correctly

### PR DESCRIPTION
**Pull request by austinabell**
Closes: #XXX

## Description

- Updates the prepare of the statedb to properly use the amino hash (was using the RLP hash for the web3 transaction type.
- Removes the hash function from that type so no confusion or misuse in the future (don't think the function can just be changed to amino hash and cache that because the tx encoder isn't known at this function)
  - I don't think this causes any problems, and the reason I was apprehensive before changing this is because signatures are derived by the RLP hash. This function was not used for any signature verification, so I assume everything will still work (Be careful though because it still uses RLP to sign and recover)
- Tx hash in the web3 api response is now correctly indicating the correct hash to index the transaction, instead of returning the Ethereum hash (might be worth looking more into to make sure dev tooling does not break because of this) 
  - Closes #176 

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/handler.go [hunk 1]
```diff
@@ -55,12 +56,13 @@ func HandleMsgEthereumTx(ctx sdk.Context, k Keeper, msg types.MsgEthereumTx) sdk
...
 		ChainID:      intChainID,
-		THash:        &txHash,
+		THash:        &ethHash,
 		Simulate:     ctx.IsCheckTx(),
 	}
+
 	// Prepare db for logs
 	// TODO: block hash
-	k.CommitStateDB.Prepare(txHash, txHash, k.TxCount)
+	k.CommitStateDB.Prepare(ethHash, common.Hash{}, k.TxCount)
```
Comment (fedekunze) [inline x/evm/handler.go:65]: can we remove the second field from the `Prepare` function signature if it's not going to be used?

**Review by fedekunze — APPROVED** [review]
tested ACK. 1 comment. Can you add a changelog entry under bug fixes?

**Review by letterj — APPROVED** [review]
LGTM

Both ethermint-scripts and ethermint-deploy ran successfully

File: x/evm/handler.go (hunk 1 above)
Comment (austinabell) [reply x/evm/handler.go:65]: it's matching the Geth interface, and will be used in the future (for adding block hash for the transactions logs)

**Pull request merged**